slice.go

```diff
@@ -282,6 +282,24 @@ func (d *Decoder) skipSlice(c byte) error {
 	return nil
 }
 
+func (d *Decoder) arrayValue(v reflect.Value) error {
+	n, err := d.DecodeBytesLen()
+	if err != nil {
+		return err
+	}
+	if n > v.Len() {
+		n = v.Len()
+	}
```

Comment:
This means that we:
- silently decode only part of the data
- leave some junk in the buffer which can break subsequent `Decode` call

What do you think if we return an error if array does not have enough capacity?